File: www/GoogleAnalyticsPlugin.js
var cordova = window.cordova || window.Cordova;
var exec = require('cordova/exec');

var GoogleAnalytics = function () {}

GoogleAnalytics.trackerWithTrackingId =  function(id, success, error) {
	if (typeof success === "function" && typeof error === "function")
		exec(success, error, "GoogleAnalyticsPlugin", "trackerWithTrackingId", [id]);
	else {
		exec(function() { console.log("trackerWithTrackingId Initialized"); },
			function() { console.log("trackerWithTrackingId isn't Initialized"); }, "GoogleAnalyticsPlugin", "trackerWithTrackingId", [id]);
	}
}

GoogleAnalytics.trackView = function(pageUri, success, error) {
	if (typeof success === "function" && typeof error === "function")
		exec(success, error, "GoogleAnalyticsPlugin", "trackView", [pageUri]);
	else {
		exec(function() { console.log("trackView Initialized"); },
			function() { console.log("trackView isn't Initialized"); }, "GoogleAnalyticsPlugin", "trackView", [pageUri]);
	}
}

/*
	category: string
	action: string
	label: string
	value: integer
*/
GoogleAnalytics.trackEventWithCategory = function(category, action, label, value, success, error) {
	var options = {
		category:category,
		action:action,
		label:label,
		value:value
	};

	if (typeof success === "function" && typeof error === "function")
		exec(success, error, "GoogleAnalyticsPlugin", "trackEventWithCategory", [options]);
	else {
		exec(function() { console.log("trackEventWithCategory Initialized"); },
			function() { console.log("trackEventWithCategory isn't Initialized"); }, "GoogleAnalyticsPlugin", "trackEventWithCategory", [options]);
	}
}

module.exports = GoogleAnalytics;

File: src/ios/GoogleAnalyticsPlugin.m
#import "GoogleAnalyticsPlugin.h"
// Dispatch period in seconds
static const NSInteger kGANDispatchPeriodSec = 2;
@implementation GoogleAnalyticsPlugin
- (void) trackerWithTrackingId:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
    NSString* accountId = [command argumentAtIndex:0];

    [GAI sharedInstance].trackUncaughtExceptions = YES;
    [GAI sharedInstance].dispatchInterval = 20;
    [GAI sharedInstance].debug = YES;
    id<GAITracker> tracker = [[GAI sharedInstance] trackerWithTrackingId:accountId];

    if (tracker == nil) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"An error occured."];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void) trackEventWithCategory:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
    NSMutableDictionary* options = (NSMutableDictionary*)[command argumentAtIndex:0];
    NSString* category = [options valueForKey:@"category"];
    NSString* action = [options valueForKey:@"action"];
    NSString* label = [options valueForKey:@"label"];
    NSNumber* value = [options valueForKey:@"value"];
    id<GAITracker> tracker = [[GAI sharedInstance] defaultTracker];
    
    @try {
        if (![tracker sendEventWithCategory:category
                            withAction:action
                             withLabel:label
                             withValue:value]) {
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"An error occured."];
        } else {
            pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
        }
    } @catch(NSException *e) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:e.name];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void) trackView:(CDVInvokedUrlCommand*)command
{
    CDVPluginResult* pluginResult = nil;
    NSString* pageUri = [command argumentAtIndex:0];
    id<GAITracker> tracker = [[GAI sharedInstance] defaultTracker];

    if (![tracker sendView:pageUri]) {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"An error occured."];
    } else {
        pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
    }
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}
@end